pkg/deploy/api/validation/validation.go

```diff
@@ -21,42 +21,56 @@ func ValidateDeploymentConfig(config *deployapi.DeploymentConfig) field.ErrorLis
 	allErrs := validation.ValidateObjectMeta(&config.ObjectMeta, true, validation.NameIsDNSSubdomain, field.NewPath("metadata"))
 
 	// TODO: Refactor to validate spec and status separately
+	specPath := field.NewPath("spec")
 	for i := range config.Spec.Triggers {
-		allErrs = append(allErrs, validateTrigger(&config.Spec.Triggers[i], field.NewPath("spec", "triggers").Index(i))...)
+		allErrs = append(allErrs, validateTrigger(&config.Spec.Triggers[i], specPath.Child("triggers").Index(i))...)
 	}
 
 	var spec *kapi.PodSpec
 	if config.Spec.Template != nil {
 		spec = &config.Spec.Template.Spec
 	}
-	specPath := field.NewPath("spec")
-	allErrs = append(allErrs, validateDeploymentStrategy(&config.Spec.Strategy, spec, field.NewPath("spec", "strategy"))...)
+	allErrs = append(allErrs, validateDeploymentStrategy(&config.Spec.Strategy, spec, specPath.Child("strategy"))...)
 	if config.Spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
 		allErrs = append(allErrs, validation.ValidatePodTemplateSpec(config.Spec.Template, specPath.Child("template"))...)
 	}
-	if config.Status.LatestVersion < 0 {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("status", "latestVersion"), config.Status.LatestVersion, "latestVersion cannot be negative"))
-	}
 	if config.Spec.Replicas < 0 {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("replicas"), config.Spec.Replicas, "replicas cannot be negative"))
 	}
 	if len(config.Spec.Selector) == 0 {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("selector"), config.Spec.Selector, "selector cannot be empty"))
 	}
+
+	statusPath := field.NewPath("status")
```

Comment:
Why not move this to a new method for `status` alone and honor the `TODO` above?